test/e2e/density.go

```diff
@@ -188,6 +219,138 @@ var _ = Describe("Density", func() {
 			// Tune the threshold for allowed failures.
 			badEvents := BadEvents(events)
 			Expect(badEvents).NotTo(BeNumerically(">", int(math.Floor(0.01*float64(totalPods)))))
+
+			if itArg.runLatencyTest {
+				Logf("Schedling additional Pods to measure startup latencies")
+
+				createTimes := make(map[string]util.Time, 0)
+				scheduleTimes := make(map[string]util.Time, 0)
+				runTimes := make(map[string]util.Time, 0)
+				watchTimes := make(map[string]util.Time, 0)
+
+				var mutex sync.Mutex
+				checkPod := func(p *api.Pod) {
+					mutex.Lock()
+					defer mutex.Unlock()
+					defer GinkgoRecover()
+
+					if p.Status.Phase == api.PodRunning {
+						if _, found := watchTimes[p.Name]; !found {
+							watchTimes[p.Name] = util.Now()
+							createTimes[p.Name] = p.CreationTimestamp
+							var startTime util.Time
+							for _, cs := range p.Status.ContainerStatuses {
+								if cs.State.Running != nil {
+									if startTime.Before(cs.State.Running.StartedAt) {
+										startTime = cs.State.Running.StartedAt
+									}
+								}
+							}
+							if startTime != util.NewTime(time.Time{}) {
+								runTimes[p.Name] = startTime
+							} else {
+								Failf("Pod %v is reported to be running, but none of its containers is", p.Name)
+							}
+						}
+					}
+				}
+
+				additionalNameStr := strconv.Itoa(minionCount) + "-" + string(util.NewUUID())
+				additionalRCName = "my-hostname-latency" + additionalNameStr
+				_, controller := framework.NewInformer(
+					&cache.ListWatch{
+						ListFunc: func() (runtime.Object, error) {
+							return c.Pods(ns).List(labels.SelectorFromSet(labels.Set{"name": additionalRCName}), fields.Everything())
+						},
+						WatchFunc: func(rv string) (watch.Interface, error) {
+							return c.Pods(ns).Watch(labels.SelectorFromSet(labels.Set{"name": additionalRCName}), fields.Everything(), rv)
+						},
+					},
+					&api.Pod{},
+					time.Millisecond*10,
```

Comment:
woah woah, this itself might dos the cluster :)
You don't want to try listing all pods every 10ms. Please make this something like 5m, the watch will still give you events and we'll still auto-relist if the connection breaks, so in theory you don't even need it.